test/relay-sproof-builder/src/lib.rs

```diff
@@ -44,14 +55,43 @@ impl Default for RelayStateSproofBuilder {
 				validation_upgrade_frequency: 6,
 				validation_upgrade_delay: 6,
 			},
+			dmq_mqc_head: None,
 			relay_dispatch_queue_size: None,
+			hrmp_ingress_channel_index: None,
 			hrmp_egress_channel_index: None,
 			hrmp_channels: BTreeMap::new(),
 		}
 	}
 }
 
 impl RelayStateSproofBuilder {
+	/// Returns a mutable reference to HRMP channel metadata for a channel (`sender`, `self.para_id`).
+	///
+	/// If there is no channel, a new default one is created.
+	///
+	/// It also updates the `hrmp_ingress_channel_index`, creating it if needed.
+	pub fn upsert_inbound_channel(&mut self, sender: ParaId) -> &mut AbridgedHrmpChannel {
```

Comment:
`upsert` I needed to google this :D But I also got no better idea.

Reply:
Rust uses `get_or_insert`, [here](https://doc.rust-lang.org/1.48.0/core/option/enum.Option.html#method.get_or_insert). Perhaps we could use it, but that seemed to me a bit overly long.

Reply:
Yeah IDK

Reply:
I also thought about this, but it would probably too long as you said.